Feature/optional filename validation

sunxyw opened the pull request:
Make filename validation more flexible for special use cases.

I have added a switch option "validateFilename" under the "editor" in .leptonrc.

e.g. In your `<homeDir>/.leptonrc`
```
"editor": {
    "tabSize": 4,
    "validateFilename": true
},
```
Filename validation will be disable when "validateFilename" switch to `false`.

Tested on Windows10, but not tested on other platforms.

hackjutsu commented on app/containers/gistEditorForm/index.js:
```diff
@@ -1,6 +1,6 @@
 import { connect } from 'react-redux'
```
Could you remove the binary file above?

hackjutsu commented on yarn.lock:
```diff
@@ -2094,7 +2094,7 @@ debug@4, debug@^4.0.1, debug@^4.1.0, debug@^4.1.1:
   dependencies:
```
I think you can also remove these yarn-xx files.

hackjutsu commented on app/containers/gistEditorForm/index.js:
```diff
@@ -84,8 +87,12 @@ class GistEditorFormImpl extends Component {
-  if (!value) return 'required'
-  else if (!validFilename(value)) return 'invalid filename'
+  if (conf.get('editor').validateFilename) {
+    if (!value) return 'required'
```
This check if the filename is empty otherwise shows "required". I don't think we should skip this check.

sunxyw commented:
I have moved the "required" check out of the the `if (conf.get('editor').validateFilename)` block.
And I have deleted some unrelated files, includes `yarn-error.log`, and the binary file.

Please feel free to tell me if it still has some problems or any improvement.

hackjutsu commented:
@sunxyw Thanks for the contributions!
